File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def bfs(n, computers, com, visited):
    visited[com] = True
    queue = deque()
    queue.append(com)
    
    while queue:
        com = queue.popleft()
        visited[com] = True
        
        for connect in range(n):
            if connect != com and computers[com][connect] == 1:
                if visited[connect] == False:
                    queue.append(connect)

def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    
    for com in range(n):
        if visited[com] == False:
            bfs(n, computers, com, visited)
            answer += 1
    
    return answer
    

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {        
    public int solution(int n, int[][] computers) {
        boolean[] visited = new boolean[n];
        int result = 0;
        
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                visited[i] = true;
                bfs(i, computers, visited);
                result++;
            }
        }
        
        return result;
    }
    
    private void bfs(int idx, int[][] computers, boolean[] visited) {
        Deque<Integer> q = new ArrayDeque<>();
        q.add(idx);
        
        while(!q.isEmpty()) {
            int nowIdx = q.poll();
            
            for (int i = 0; i < computers.length; i++) {
                if (i != nowIdx && computers[nowIdx][i] == 1) {
                    if (!visited[i]) {
                        visited[i] = true;
                        q.add(i);
                    }
                }
            }
        }
    }
}